### PR TITLE
Driveinfo work

### PR DESCRIPTION
**Pull request by fanoI**
This PR adds support for DriveInfo that is the starting point of Cosmos DiskManager (#924).

Another important thing that is changed is how VFS check what filesystem a partition has using a registration mechanism: this open the possibility to create more Filesystem for Cosmos.

### CODE REVIEW

File: source/Cosmos.System2/FileSystem/FileSystem.cs [hunk 1]
```diff
@@ -45,5 +35,13 @@ public static FileSystemType GetFileSystemType(Partition aDevice)
         public string mRootPath { get; }
 
         public long mSize { get; }
+
+        public abstract long mAvailableFreeSpace { get; }
```
Comment (jp2masa) [inline source/Cosmos.System2/FileSystem/FileSystem.cs]: Property names should be pascal cased (in this case `AvailableFreeSpace`).

**Comment by fanoI** [comment]
In reality the file in which the file is contained is "FatDiretoryEntry.cs" I wanted to change it to "FatDirectoryEntry.cs" but I've noticed that then Github would seen the file totally changed and this would "obstruct" the review. I don't have noticed that instead I've changed the class name there, I'll change it.

Regarding "mAvailableFreeSpace" I think all this class has similar problems for example there are other properties that seem "members" (they start with "m"), maybe better to change them all? For example instead  mSize => Size, mLabel => Label and so on?

**Comment by jp2masa** [comment]
>I wanted to change it to "FatDirectoryEntry.cs" but I've noticed that then Github would seen the file totally changed and this would "obstruct" the review.

In those cases, you can rename the file in a separate commit (no need for a PR).

>maybe better to change them all?

I think so, we should use the standard naming conventions for public APIs.

**Comment by charlesbetros** [comment]
I would verify with Chad regarding proper naming of properties, fields, locals, params. There are some specific rules. Not sure about public properties, but the others have prefixes we follow.

**Comment by czhower** [comment]
Im writing up a doc now. Will post soon regarding naming.

**Pull request merged**